client/components/domains/use-my-domain/utilities/connect-domain-action.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import page from 'page';
+/**
+ * Internal dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+import { domainManagementList, domainMappingSetup } from 'calypso/my-sites/domains/paths';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import isSiteOnPaidPlan from 'calypso/state/selectors/is-site-on-paid-plan';
+
+export const connectDomainAction = ( { domain, selectedSite } ) => ( dispatch, getState ) => {
+	const siteHasPaidPlan = isSiteOnPaidPlan( getState(), selectedSite.ID );
+
+	if ( selectedSite.is_vip ) {
+		wpcom
+			.site( selectedSite.ID )
+			.addVipDomainMapping( domain )
+			.then( () => page( domainManagementList( selectedSite.slug ) ) )
+			.catch( ( error ) => dispatch( errorNotice( error.message ) ) );
+	} else if ( siteHasPaidPlan ) {
+		wpcom
+			.site( selectedSite.ID )
+			.addDomainMapping( domain )
+			.then( () => {
+				dispatch(
+					successNotice(
+						__( 'Domain connected! Please make sure to follow the next steps below.' ),
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Domain mapping added! Please make sure to follow the next steps below.' )` *ES Score: 7*